src/Util/ClassNameDetails.php

```diff
@@ -15,13 +15,13 @@
 
 final class ClassNameDetails
 {
-    private $fullClassName;
-    private $namespacePrefix;
-    private $suffix;
+    private ?string $suffix;
 
-    public function __construct(string $fullClassName, string $namespacePrefix, string $suffix = null)
-    {
-        $this->fullClassName = $fullClassName;
+    public function __construct(
+        private string $fullClassName,
+        private string $namespacePrefix,
+        string $suffix = null
```

Comment:
```suggestion
        private ?string $suffix = null
```

and the member and assignment can be removed?